### PR TITLE
Modify Cython api to use `api` keyword 

### LINKED ISSUE #1779: Cython API is not implemented correctly
We still rely on importing our three cython modules in `__init__.py` to fix the segfaults at the end of `cyclus.main()`.  Now in macos builds I am experiencing the same segfaults, even with those imports.  A proper/correct fix to the python interpreter management should be found.

### PR DESCRIPTION
**Pull request by bennibbelink**
I stumbled upon [this section ](https://cython.readthedocs.io/en/latest/src/userguide/external_C_code.html#c-api-declarations)of the cython docs (I can't believe I missed it before!!) that discusses how to make cython declarations available to C code.  Since we only access cython methods in PyHooks we can define those methods with the `api` keyword and avoid dealing with the messy multi-phase module initialization (see #1781).  It also does not require that `libcyclus` is linked with our cython modules since "the Python import machinery is used to make the connection dynamically" (if you look at the generated `_api.h` headers it looks like they are doing a lot of the messy Python C-API calls that I was making in #1781, but in a robust/correct way :smile:).

This PR replaces #1781 and closes #1779.

### CODE REVIEW

File: cyclus/__init__.py [hunk 1]
```diff
@@ -1,2 +1,2 @@
 __version__ = 'x.x.x' # will be populated during build
-import pymodule, eventhooks, pyinfile
\ No newline at end of file
+# import pymodule, eventhooks, pyinfile
```
Comment (gonuke) [inline cyclus/__init__.py:2]: Should we just delete this line?

**Review by gonuke — CHANGES_REQUESTED** [review]
Thanks @bennibbelink - I'm not sure I understand all of this, but probably didn't understand it well enough before, either.  Just one little comment here.

File: cyclus/__init__.py (hunk 1 above)
Comment (bennibbelink) [reply cyclus/__init__.py]: Yes, thanks for catching that

**Review by gonuke — APPROVED** [review]
Thanks @bennibbelink

**Pull request merged**